### ISSUE TITLE
Feature Request: Listenbrainz integration

### ISSUE THREAD

**Issue by RebelMoogle** [ytmdesktop/ytmdesktop#257]
Would be great to have a Listenbrainz integration in addition to Last.fm integration.

**Comment by Nassoo** [comment]
Yes, please! It would be wonderfull addition to your great product!

**Comment by zjvandermissen** [comment]
A +1 to this as an excellent feature, listenbrainz is open in giving you access to your own data, more so than last.fm

**Comment by AnnanFay** [comment]
Is there any way right now to scrobble from Youtube Music to ListenBrainz on desktop?

Google Play Music Desktop Player had scrobbling but is broken for a few months. Probably because all music comes up as "Unknown Song - Unknown Artist". ytmdesktop looks like it's more up to date, but this topic hasn't been replied to since July.

I'd offer to help implementation, but no idea where to start.

The other app is MIT License so _might_ be simple to port over. I think relevant code is in:

* [./src/main/features/core/listenBrainz.js](https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/blob/70db2a0c0bee8d4c7b21c24a9caad23d036230d7/src/main/features/core/listenBrainz.js)
* [./src/renderer/ui/components/settings/tabs/ListenBrainzTab.js](https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/blob/70db2a0c0bee8d4c7b21c24a9caad23d036230d7/src/renderer/ui/components/settings/tabs/ListenBrainzTab.js)

**Comment by kabili207** [comment]
> Is there any way right now to scrobble from Youtube Music to ListenBrainz on desktop?

Youtube Music is a progressive web app (PWA) so you could install it that way, although you'd lose all the extras that YTMDesktop adds. If you're on Linux you can use something like [mpris-scrobbler](https://github.com/mariusor/mpris-scrobbler).

**Comment by jippylong12** [comment]
Put together a PR (#1706) that adds ListenBrainz scrobbling as an integration alongside Last.fm. It follows the same architecture as the existing Last.fm integration:

- Toggle in Settings > Integrations
- Token-based auth (paste your token from listenbrainz.org/settings/)
- Now-playing updates and scrobbling with configurable scrobble percent
- Token encrypted via safeStorage

For anyone looking for a workaround in the meantime, the [Web Scrobbler](https://web-scrobbler.com/) browser extension supports ListenBrainz with YouTube Music in a browser, or on Linux you can use [mpris-scrobbler](https://github.com/mariusor/mpris-scrobbler) as @kabili207 mentioned.

@rickpalmeira
